File: backend/Models/Transaction.cs
namespace FinTrack.Api.Models;

public class Transaction
{
    public Guid Id { get; set; }
    public string Type { get; set; } = "expense"; // income, expense
    public decimal Amount { get; set; }
    public Guid AccountId { get; set; }
    public Account Account { get; set; } = null!;
    public Guid CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public string Note { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
}

File: src/types/transaction.ts
export interface Transaction {
  id: string;
  type: "income" | "expense";
  amount: number;
  accountId: string;
  accountName?: string;
  categoryId: string;
  categoryName?: string;
  categoryColor?: string;
  note: string;
  date: string;
}

export interface TransactionFormData {
  type: "income" | "expense";
  amount: number;
  accountId: string;
  categoryId: string;
  note: string;
  date: string;
}
